vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java

```diff
@@ -364,6 +366,38 @@ public void reroute(HttpMethod method, String path) {
     return getPathParams().get(name);
   }
 
+  @Override
+  public Map<String, String> queryParams() {
+    return getQueryParams();
+  }
+
+  @Override
+  public @Nullable List<String> queryParam(String query) {
+    return getQueryParamsParsed().get(query);
+  }
+
+  protected void setQueryParams(Map<String, List<String>> q) {
+    queryParamsParsed = new HashMap<>();
+    queryParamsParsed.putAll(q);
+    queryParams = new HashMap<>();
+    for (Map.Entry<String, List<String>> e : q.entrySet())
+      queryParams.put(e.getKey(), String.join(",", e.getValue()));
+  }
+
+  private Map<String, String> getQueryParams() {
+    if (queryParams == null) {
+      queryParams = new HashMap<>();
+    }
+    return queryParams;
+  }
+
+  private Map<String, List<String>> getQueryParamsParsed() {
```

Comment:
This becomes not needed if we populate from the base object and parse on `setPathParams` right?